### PR TITLE
added simple root Event and fixed OGBillOfLading Generator

### PR DESCRIPTION
**Pull request by mkhraisha**
Adds a base "Event" Schema that captures very simple information. 

This schema should be extended by the following EPCIS event types:
* observation
* transformation
* aggregation
* disaggregation
* commission
* decommission

This PR is meant to spark conversation on what fields need to be in the base Event vs one of the Event Types

### CODE REVIEW

File: packages/traceability-schemas/schemas/Event.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            "description": "Event Identifier.",
+            "type": "string"
+        },
+        "actor": {
+            "$comment": "{\"term\": \"actor\", \"@id\": \"https://w3id.org/traceability#Organization\", \"@type\": \"https://schema.org/ItemList\"}",
+            "title": "Actor",
+            "description": "The organization performing the activity.",
+            "type": "array",
+            "items": {
+                "$ref": "https://w3id.org/traceability/schemas/Organization.json"
```
Comment (mprorock) [inline packages/traceability-schemas/schemas/Event.json:40]: Do we want this to be an org? or is entity possibly better?  We at least also deal with individuals in the supply chain where Person sometimes is a better representation than Organization

File: packages/traceability-schemas/schemas/Event.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            "description": "Location where event took place",
+            "$ref": "https://w3id.org/traceability/schemas/Place.json"
+        },
+        "eventTime": {
+            "$comment": "{\"term\": \"eventTime\", \"@id\": \"https://schema.org/DateTime#v2\"}",
+            "title": "Event Time",
+            "description": "Time when the event took place",
+            "type": "string"
+        },
+        "products": {
```
Comment (mprorock) [inline packages/traceability-schemas/schemas/Event.json:55]: I am going to have to mull over use of Product here.  I think it is probably correct, but also thinking about raw materials becoming a product etc.

File: packages/traceability-schemas/schemas/Event.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                "$ref": "https://w3id.org/traceability/schemas/Organization.json"
+            }
+        },
+        "eventLocation": {
+            "$comment": "{\"term\": \"eventLocation\", \"@id\": \"https://w3id.org/traceability#eventLocation\"}",
+            "title": "Event Location",
+            "description": "Location where event took place",
+            "$ref": "https://w3id.org/traceability/schemas/Place.json"
+        },
+        "eventTime": {
```
Comment (mprorock) [inline packages/traceability-schemas/schemas/Event.json:49]: we may want to actually record a few more things here (possibly optionally), e.g. scheduled or planned start time, planned duration, actual start time, actual end time, etc.  open on this.

**Review by mprorock — CHANGES_REQUESTED** [review]
Awesome start on this - I think the Event is going to be a a key interop point, and we will certianly need to issue VCs against certain events at least in our case on the ag side of things

**Comment by OR13** [comment]
@mprorock  can you clearly state what changes need to be made in order to accept this PR?

potentially we can open issues to track them and iterate?

File: packages/traceability-schemas/schemas/Event.json [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            "$comment": "{\"term\": \"actor\", \"@id\": \"https://w3id.org/traceability#Organization\", \"@type\": \"https://schema.org/ItemList\"}",
+            "title": "Actor",
+            "description": "The organization performing the activity.",
+            "type": "array",
+            "items": {
+                "$ref": "https://w3id.org/traceability/schemas/Organization.json"
+            }
+        },
+        "eventLocation": {
+            "$comment": "{\"term\": \"eventLocation\", \"@id\": \"https://w3id.org/traceability#eventLocation\"}",
```
Comment (OR13) [inline packages/traceability-schemas/schemas/Event.json:44]: I am not sure we want to be defining "eventLocation"... can we use an existing ontology for this field?

File: packages/traceability-schemas/schemas/Event.json [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            "$ref": "https://w3id.org/traceability/schemas/Place.json"
+        },
+        "eventTime": {
+            "$comment": "{\"term\": \"eventTime\", \"@id\": \"https://schema.org/DateTime#v2\"}",
+            "title": "Event Time",
+            "description": "Time when the event took place",
+            "type": "string"
+        },
+        "products": {
+            "$comment": "{\"term\": \"products\", \"@id\": \"https://www.gs1.org/voc/Product\"}",
```
Comment (OR13) [inline packages/traceability-schemas/schemas/Event.json:56]: noting the use of a GS1 vocabulary item here, which is close to but not equal to schema.org product.

File: packages/traceability-schemas/src/generators/Event.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+const faker = require('faker');
+const { getOrganization } = require('./Organization');
+const { getPlace } = require('./Place');
+const { getProduct } = require('./Product');
+
+const getEvent = () => {
+    const eventType = 'commission';
+    const eventId = '12345';
+    const actor = [getOrganization(), getOrganization()];
```
Comment (OR13) [inline packages/traceability-schemas/src/generators/Event.js:9]: likely these helpers are injecting `@context`... please make sure we are not testing them in example data.

**Review by OR13 — CHANGES_REQUESTED** [review]
https://github.com/w3c-ccg/traceability-vocab/pull/164/files#r634604678

^ this is the only critical objection I have, we need to be cleaning up our examples and avoid context nesting.

File: packages/traceability-schemas/schemas/Event.json (hunk 4 above)
Comment (mprorock) [reply packages/traceability-schemas/schemas/Event.json:44]: +1 I think this should be a [Place](https://w3c-ccg.github.io/traceability-vocab/#Place)

File: packages/traceability-schemas/schemas/Event.json (hunk 2 above)
Comment (mprorock) [reply packages/traceability-schemas/schemas/Event.json:55]: see note from @OR13 below - should this be GS1 or schema.org?  Other aspects of the vocab are utilizing schema.org

File: packages/traceability-schemas/schemas/Event.json (hunk 3 above)
Comment (mprorock) [reply packages/traceability-schemas/schemas/Event.json:49]: I would prefer ability to record at a minimum

1. event start time
2. event end time

File: packages/traceability-schemas/schemas/Event.json (hunk 1 above)
Comment (mkhraisha) [reply packages/traceability-schemas/schemas/Event.json:40]: I had misgivings on whether we should have two fields or just one for this, I wanted initially to have an `actor` and an `actingOrg` field. 

An entity might solve the problem. good shout

Comment (mprorock) [reply packages/traceability-schemas/schemas/Event.json:40]: I would prefer use of [Entity](https://w3c-ccg.github.io/traceability-vocab/schemas/Entity.json)

File: packages/traceability-schemas/schemas/Event.json (hunk 5 above)
Comment (mprorock) [reply packages/traceability-schemas/schemas/Event.json:56]: > @mprorock can you clearly state what changes need to be made in order to accept this PR?
> 
> potentially we can open issues to track them and iterate?

noted above

File: packages/traceability-schemas/schemas/Event.json (hunk 3 above)
Comment (mkhraisha) [reply packages/traceability-schemas/schemas/Event.json:49]: I think this makes sense to split up, probably another base object that encapsulates eventTime that would have the planned start/duration etc.

File: packages/traceability-schemas/schemas/Event.json (hunk 4 above)
Comment (mkhraisha) [reply packages/traceability-schemas/schemas/Event.json:44]: absolutely should be place

File: packages/traceability-schemas/src/generators/Event.js (hunk 6 above)
Comment (mkhraisha) [reply packages/traceability-schemas/src/generators/Event.js:9]: theres two ways to fix this in my mind either:
 * we need a parameter to these helpers that indicates whether to inject context or not (my preference)
 * we need to strip context after its generated in each of these helpers

**Review by mprorock — APPROVED** [review]

**Comment by mkhraisha** [comment]
Removed child context @mprorock  i deleted the generator and addressed most things in this conversation except for the event time stuff. would it be ok if i open an issue and assign it to you?

**Comment by mprorock** [comment]
> event time stuff

let's not worry about the event time stuff as a blocker for now - we can add that as a later detail when required - probably as a subclass of this (e.g. event with duration or something like that)

**Review by OR13 — APPROVED** [review]

**Comment by OR13** [comment]
On the call we discussed giving this one for day for reviews.

File: packages/traceability-schemas/schemas/Event.json [hunk 7]
```diff
@@ -0,0 +1,67 @@
+{
+    "$id": "https://w3id.org/traceability/schemas/Event.json",
+    "$schema": "https://json-schema.org/draft-07/schema#",
+    "$comment": "{\"term\": \"Event\", \"@id\": \"https://schema.org/Event\"}",
+    "title": "Organization",
+    "description": "An event such as transformation, aggregation, commission etc.",
```
Comment (TallTed) [inline packages/traceability-schemas/schemas/Event.json:6]:
```suggestion
    "description": "An event such as transformation, aggregation, commission, etc.",
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Made many description strings more consistent with each other (start with article, close with period, etc.), particularly in ontology.  Some sample data is not good English (sentence fragments, primarily) but I'm not always sure what was intended, so some are now unchanged — but will want followup attention.

**Comment by mkhraisha** [comment]
The remaining suggestions modify generated text, we probably need to actually dive into the generators and figure out how to fix them. This should probably be made its own issue and tracked accordingly. @OR13 @mprorock @TallTed

**Comment by OR13** [comment]
These proposed changed look good to me, @mkhraisha can you approve them / reject them so we can see where any gaps are remaining?

File: packages/traceability-schemas/schemas/Event.json (hunk 2 above)
Comment (TallTed) [reply packages/traceability-schemas/schemas/Event.json:55]: Consider "inputs" (raw materials, outputs from other processes/sources, etc.) and "outputs" (refined version of inputs becoming inputs to other processes/consumers, finished products, etc.).  These can then be easily turned into a complex *trace* from most-basic-origin (e.g., extraction of various ores from deep underground) to most-refined-consumable (e.g., a smartphone, tablet, or computer)...

**Comment by OR13** [comment]
Feels like the generators may be getting in the way of having "realistic fixtures"... we should continue to discuss this on the related open issues.

**Comment by TallTed** [comment]
Generated files should generally not be in git repos.  Whatever can be done to shift things so that the generated files go to github.io or similar would be best.

**Comment by mkhraisha** [comment]
Apologies for the delays in tackling this. 
 Should be all good to go now.

**Comment by OR13** [comment]
This looks good to me.

**Pull request merged**